### PR TITLE
LPS-153119 Deprecate `Liferay.provide('_openWindowProvider')`

### PR DESCRIPTION
**Pull request by kresimir-coko**
I've moved the implementation of `_openWindowProvider` into `openWindow` because that is the only place where it is used. Because of this, it is possible to deprecate the `_openWindowProvider` 👍 

It's debatable if we can just straight up delete `_openWindowProvider` because it is a "private" method and it shouldn't be used by anyone by us, right?

Can be tested by dragging a Calendar portlet to the landing page and creating a new event.

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js [hunk 1]
```diff
@@ -532,11 +532,11 @@
 		openWindow(config, callback) {
 			config.openingWindow = window;
 
-			var top = Util.getTop();
-
-			var topUtil = top.Liferay.Util;
+			var dialog = Window.getWindow(config);
```
Comment (bryceosterhaus) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js:535]: I think for this bit of code to work properly, the `liferay-util-window` module must already be loaded, which is why `_openWindowProvider` is provided in a `Liferay.provide` with a dependency listed [here](https://github.com/liferay-frontend/liferay-portal/pull/2205/files#diff-22ee0ee3757e1d50cbee0814c51e24fd2d987a91091b54dde287c3b7034ec30cL1403).

**Review by bryceosterhaus — CHANGES_REQUESTED** [review]

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js (hunk 1 above)
Comment (kresimir-coko) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js:535]: `Liferay.Util.Window` is accessible either way, isn't that the "only" thing we care about here?

I mean, I've tested this and it works just as it did before the change.

Comment (kresimir-coko) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js:535]: I see, so this needs to be done within the `Liferay.provide` deprecation project. Close the PR if you agree.

**Comment by kresimir-coko** [comment]
Closing because it should be done as part of a future epic where we modernize `Liferay.provide`

**Pull request closed without merging**